lib/rack/cache/storage.rb

```diff
@@ -38,6 +38,9 @@ def create_store(type, uri)
         else
           fail "Unknown storage provider: #{uri.to_s}"
         end
+      elsif uri.nil?
+        # hack to support setting entitystore to nil, which doesn't persist request bodies
+        type.const_get('NIL').resolve(uri)
```

Comment:
All other subclasses of EntityStore are instanced like this, calling ```Foo.resolve(uri)``` which returns the new instance. There is no call to ```Foo.new``` to instance an entity store in ```Storage.create_store```. I just followed the same style.

If what you dislike is the use if the magic string ```'NIL'``` I can change it to this: 

```
elsif uri.nil?
        # hack to support setting entitystore to nil, which doesn't persist request bodies
        type.const_get(uri.inspect.upcase).resolve(uri)
```

Better?